File: app/src/main/java/com/neocaptainnemo/cv/ui/ArrayAdapter.java
package com.neocaptainnemo.cv.ui;

import android.content.Context;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public abstract class ArrayAdapter<T, V extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<V> {

    protected List<T> data = new ArrayList<>();
    protected Context context;

    protected ArrayAdapter(Context context) {
        this.context = context;
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public void clear() {
        data.clear();
    }

    public void add(Collection<T> dataToAdd) {
        data.addAll(dataToAdd);
    }

    public void add(T item) {
        data.add(item);
    }
}

File: app/src/main/java/com/neocaptainnemo/cv/ui/ArrayAdapter.kt
package com.neocaptainnemo.cv.ui

import androidx.recyclerview.widget.RecyclerView
import java.util.*

abstract class ArrayAdapter<T, V : androidx.recyclerview.widget.RecyclerView.ViewHolder>() : androidx.recyclerview.widget.RecyclerView.Adapter<V>() {

    protected var data: MutableList<T> = ArrayList()

    override fun getItemCount(): Int = data.size

    fun clear() = data.clear()

    fun add(dataToAdd: Collection<T>) = data.addAll(dataToAdd)

    fun add(item: T) = data.add(item)
}
